src/main/java/org/qiyana/DAO/imp/ProductDao.java

```diff
@@ -0,0 +1,88 @@
+package org.qiyana.DAO.imp;
+
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.qiyana.DAO.Dao;
+import org.qiyana.model.Product;
+
+/** Class responsible for database operations at product entity. */
+@Slf4j
+public class ProductDao implements Dao<Product> {
+  EntityManager em;
+
+  public ProductDao(EntityManagerFactory etf) {
+    em = etf.createEntityManager();
+  }
+
+  /**
+   * Uses an entity manager to insert a product into the database.
+   *
+   * @param product the product object who will be added at database.
```

Comment:
will be added *to* the database

`at` is used for locations/positions (I'm at the restaurant, I'm at the rooftop, etc)